packages/p2p-v2/src/components/Table/Table.tsx

```diff
@@ -33,23 +34,49 @@ const Table = <T,>({
     });
 
     const tableContainerRef = useRef<HTMLDivElement>(null);
+    const headerRef = useRef<HTMLDivElement>(null);
+    const [height, setHeight] = useState(0);
+
+    useLayoutEffect(() => {
+        if (headerRef?.current) {
+            const topPosition = headerRef.current.getBoundingClientRect().bottom;
+            setHeight(window.innerHeight - topPosition);
+        }
+    }, [headerRef?.current]);
+
     const { fetchMoreOnBottomReached } = useFetchMore({
         loadMore: loadMoreFunction,
         ref: tableContainerRef,
         isFetching,
     });
 
     return (
-        <div
-            className={clsx('p2p-v2-table', tableClassname)}
-            onScroll={e => fetchMoreOnBottomReached(e.target as HTMLDivElement)}
-            ref={tableContainerRef}
-        >
-            {table.getRowModel().rows.map(row => (
-                <div className={rowClassname} key={row.id}>
-                    {rowRender(row.original)}
+        <div className='w-full'>
+            {isDesktop && columns.length > 0 && (
+                <div
+                    className='p2p-v2-table__header'
+                    ref={headerRef}
+                    style={{ gridTemplateColumns: `repeat(${columns.length}, 1fr)` }}
+                >
+                    {table.getFlatHeaders().map(header => (
+                        <Text className='p2p-v2-table__header-items' key={header.id} size='sm' weight='bold'>
+                            {headerRender(header.column.columnDef.header as string)}
+                        </Text>
```

Comment:
This is just an opinion and another way to do it, soo ignore this if this is too complex:
In this case, why not just pass the Header component itself, something like `renderHeader={MyHeader}` and then use it inside the component like:
```
<MyHeader>{header.column.columnDef.header}</MyHeader>
```
or alternatively, you can use compound components to construct this table, so we can declaratively use the headers like:
```
<Table>
   <Table.Headers as={MyHeader} headers={['header 1', 'header 2', ...]} />
   <Table.Body>
      <Table.Row>...</Table.Row>
       <Table.Row>...</Table.Row>
   </Table.Body>
```